src/vm-vcpu/src/vm.rs

```diff
@@ -229,7 +237,10 @@ impl<EH: 'static + ExitHandler + Send> KvmVm<EH> {
         let mut vm = Self::create_vm(kvm, vm_config, exit_handler, guest_memory)?;
 
         #[cfg(target_arch = "x86_64")]
-        MpTable::new(vm.config.num_vcpus)?.write(guest_memory)?;
+        {
+            let max_irq:u8 = vm.config.max_irq.try_into().expect("max irq should be u8");
```

Comment:
We shouldn't use expect here. Let's return an error instead. We should also add a negative test here to check that when passing a IRQ num > u8::MAX the code does not panic and rather returns an error.